src/box/applier.cc

```diff
@@ -1252,6 +1298,726 @@ applier_on_rollback(struct trigger *trigger, void *event)
 	return 0;
 }
 
+/** The underlying thread behind a number of appliers. */
+struct applier_thread {
+	/** A link in allocated threads list. */
+	struct rlist in_list;
+	struct cord cord;
+	/** The single thread endpoint. */
+	struct cbus_endpoint endpoint;
+	/** A pipe from the applier thread to tx. */
+	struct cpipe tx_pipe;
+	/** A pipe from tx to the applier thread. */
+	struct cpipe thread_pipe;
+	/** A conditional variable to wait on until thread initialization. */
+	struct fiber_cond ready_cond;
+	/**
+	 * A notification set for tx once the thread is operational: the
+	 * endpoint and both pipes are created.
+	 */
+	bool is_ready;
```

Comment:
`is_ready` and `ready_cond` are not needed AFAIU - `cpipe_create` blocks until the other endpoint is created.

Reply:
Yes, but I call cbus_pair from the thread. So I can't guarantee that `cpipe_create` would be invoked before the first applier tries to push a message to the pipe.

Am I missing something obvious?

Reply:
I see. Maybe, not use `cbus_pair` at all then and create the tx -> applier pipe in tx, like it's done in, for example, `vy_worker_pool_start`?